org/postgresql/test/jdbc2/ArrayTest.java

```diff
@@ -413,6 +413,26 @@ public void testNonStandardDelimiter() throws SQLException
 
         assertTrue(!arrRS.next());
     }
+    
+    public void testEmptyBinaryArray() throws SQLException
+    {
+        
+        PreparedStatement pstmt = conn.prepareStatement("SELECT '{}'::int[]");
+        // force binary
+        ((org.postgresql.PGStatement)pstmt).setPrepareThreshold(-1);
+        
+        ResultSet rs = pstmt.executeQuery();
+ 
+        while (rs.next()) {
+            Array array = rs.getArray(1);
+            if (!rs.wasNull())
+            {
+                ResultSet ars  = array.getResultSet();
+                assertEquals(java.sql.Types.INTEGER,ars.getMetaData().getColumnType(1)); 
```

Comment:
Having explicit message in `assertEquals` is a big plus as it would explain failure without any need to look up the code:  `ars.getMetaData().getColumnType(1) should return Types.INTEGER, expected: X, actual: Y` is much more readable than `expected: X, actual: Y`